File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        # kadanes algorithm
        curr = nums[0]
        maxi = nums[0]

        for i in nums[1:]:
            curr = max(i, curr+i)
            maxi = max(maxi, curr)
        
        return maxi

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        // kadanes algorithm
        int c=0, m=INT_MIN;
        for(int i=0;i<nums.size();i++){
            c = max(nums[i], nums[i]+c);
            m = max(m, c);
        }
        return m;
        
    }
};